src/util.c

```diff
@@ -444,12 +444,12 @@ int write_result_into_xml_file(struct ntttcp_test_endpoint *tep)
 		fprintf(logfile, "	<tcp_average_rtt metric=\"us\">%u</tcp_average_rtt>\n", tepr->average_rtt);
 	}
 
-	os_info = malloc(256);
-	memset(os_info, '\0', sizeof(char) * 256);
-	count = execute_system_cmd_by_process("uname -a", "r", os_info);
-	escape_char_for_xml(os_info, os_info_escaped);
+	count = execute_system_cmd_by_process("uname -a", "r", &os_info);
+	if (os_info) {
+		escape_char_for_json(os_info, os_info_escaped);
```

Comment:
do we really need this "escape_char_for_json"? looks like we only use it to remove the "\n" in the string.

Reply:
I'm sorry, there should is "escape_char_for_xml". I have been modified.